### PR TITLE
Special-case DNS client ports in SOCKS5-UDP

### PR DESCRIPTION
**Pull request by bemasc**
This reduces the typical lifetime of a SOCKS5 UDP Association
for DNS from 10 seconds (the idle timeout) to under 100 milliseconds.
That should reduce the average DNS-related memory usage by a factor
of over 100.

### CODE REVIEW

File: third_party/badvpn/socks_udp_client/SocksUdpClient.c [hunk 1]
```diff
@@ -424,6 +452,14 @@ static int compute_mtu(int udp_mtu)
     return udp_mtu + sizeof(struct socks_udp_header) + sizeof(struct socks_addr_ipv6);
 }
 
+static int dns_id(BAddr *remote_addr, const uint8_t *data, int data_len)
```
Comment (alalamav) [inline third_party/badvpn/socks_udp_client/SocksUdpClient.c]: nit: rename to use a verb, i.e. `get_dns_id`

Comment (alalamav) [reply third_party/badvpn/socks_udp_client/SocksUdpClient.c]: Could you leverage this implementation to extract the query ID? https://github.com/Jigsaw-Code/outline-client/blob/master/third_party/badvpn/misc/dns_proto.h#L54

Comment (bemasc) [reply third_party/badvpn/socks_udp_client/SocksUdpClient.c]: Done

**Comment by trevj** [comment]
@bemasc Cool! Can you please update `METADATA`?

@alalamav My understanding is this won't do anything for Apple - should we make the change there, too?

**Comment by bemasc** [comment]
OK, I added a mention of this change in `third_party/badvpn/METADATA`.

**Comment by alalamav** [comment]
@trevj, this change is not necessary in Apple, since DNS queries are processed through a dedicated socket (like Android).

**Comment by trevj** [comment]
@bemasc Thanks!

@alalamav OK, good to know.

**Pull request merged**